mobile-app/app/screens/AppNavigator/screens/Balances/hooks/TokenPrice.ts

```diff
@@ -1,35 +1,17 @@
-import { useCallback, useMemo } from 'react'
+import { useCallback } from 'react'
 import BigNumber from 'bignumber.js'
 import { RootState } from '@store'
-import { PoolPairData } from '@defichain/whale-api-client/dist/api/poolpairs'
 import { useSelector } from 'react-redux'
-import { checkIfPair, findPath, GraphProps } from '@screens/AppNavigator/screens/Dex/helpers/path-finding'
-import { CacheApi } from '@api/cache'
-import { useNetworkContext } from '@shared-contexts/NetworkContext'
-
-interface CalculatePriceRatesI {
-  aToBPrice: BigNumber
-  bToAPrice: BigNumber
-  estimated: BigNumber
-}
 
 interface TokenPrice {
   getTokenPrice: (symbol: string, amount: BigNumber, isLPS?: boolean) => BigNumber
-  calculatePriceRates: (fromTokenSymbol: string, pairs: PoolPairData[], amount: BigNumber) => CalculatePriceRatesI
-  getArbitraryPoolPair: (tokenASymbol: string, tokenBSymbol: string) => PoolPairData[]
 }
 
-export function useTokenPrice (): TokenPrice {
-  const { network } = useNetworkContext()
+export function useTokenPrice (denominationTokenSymbol = 'USDT'): TokenPrice {
   const blockCount = useSelector((state: RootState) => state.block.count)
+  const dexPrices = useSelector((state: RootState) => state.wallet.dexPrices)
```

Comment:
You can just make this a full selector. A selector that gets prices based on denominationTokenSymbol, so it's memoized. https://github.com/reduxjs/reselect